File: examples/docs-code-snippets/src/main/scala/examples/scala/recipes/RecipeWithReproviderInteraction.scala
package examples.scala.recipes

import com.ing.baker.recipe.scaladsl.{Event, Recipe}
import examples.scala.events.OrderPlaced
import examples.scala.interactions.ShipOrder

object RecipeWithReproviderInteraction {
  val recipe: Recipe = Recipe("Reprovider recipe")
    .withSensoryEvent(Event[OrderPlaced])
    .withInteraction(
      ShipOrder.interaction
        .isReprovider(true)
        .withRequiredEvent(Event[OrderPlaced])
    )
}

File: examples/docs-code-snippets/src/main/java/examples/java/recipes/RecipeWithReproviderInteraction.java
package examples.java.recipes;

import com.ing.baker.recipe.javadsl.InteractionDescriptor;
import com.ing.baker.recipe.javadsl.Recipe;
import examples.java.events.OrderPlaced;
import examples.java.interactions.ShipOrder;

public class RecipeWithReproviderInteraction {

    public final static Recipe recipe = new Recipe("Reprovider recipe")
        .withSensoryEvent(OrderPlaced.class)
        .withInteractions(
            InteractionDescriptor.of(ShipOrder.class)
                    .isReprovider(true)
                    .withRequiredEvent(OrderPlaced.class)
        );
}
